imap-codec/src/codec/encode.rs

```diff
@@ -325,16 +325,34 @@ impl<'a> EncodeIntoContext for CommandBody<'a> {
                 ctx.write_all(b" ")?;
                 password.declassify().encode_ctx(ctx)
             }
-            CommandBody::Select { mailbox } => {
+            CommandBody::Select { mailbox, parameters } => {
                 ctx.write_all(b"SELECT")?;
                 ctx.write_all(b" ")?;
-                mailbox.encode_ctx(ctx)
+                mailbox.encode_ctx(ctx)?;
+                if let Some(p) = parameters {
+                    ctx.write_all(b" (")?;
+                    for atom in p.as_ref().iter() {
+                        atom.encode_ctx(ctx)?;
```

Comment:
Just because I saw it: I think there is whitespace missing between the atoms? Maybe use `join_serializable`?